grep-00XX-relicense-volk.md

```diff
@@ -0,0 +1,141 @@
+# GREP [00XX] -- [Re-license VOLK under LGPL]
+
+- Original Author: [Johannes Demel <demel@uni-bremen.de>]
+- Champion: [Johannes Demel <demel@uni-bremen.de>]
+- Status: Draft
+
+History:
+- 18-Mar-2021: Initial Draft
+
+## Abstract
+
+We want to remove usage entry barriers from VOLK.
+As a result, we expect greater adoption and a growing user and contributor base of VOLK.
+This move helps to spread the value of Free and Open Source Software in the SIMD community, wich so far is dominated by non-FOSS software.
+The proposed new license is [LGPLv3+](https://www.gnu.org/licenses/lgpl-3.0.html).
+Historically, VOLK was part of GNU Radio. Thus, it made total sense to use GPLv3+ just like GNU Radio.
+Since then, VOLK became its own project with its own repository and leadership.
+While it is still a core dependency of GNU Radio and considers GNU Radio as its main user, others may want to use it too.
+Especially, long term VOLK contributors expressed their desire to be able to use VOLK in a broader set of projects.
+
+### State of the GREP
+
+This GREP currently serves as a platform to collect all the necessary information and ideas to relicense VOLK.
+Everyone is invited to contribute to this GREP with ideas, pull requests, comments etc.
+
+## Copyright / License
+
+This GREP is licensed as CC-BY-ND.
+Copyright 2021 Johannes Demel and any co-author listed above.
+
+
+## Motivation
+
+We recognize the desire of our long term contributors to be able to use VOLK with their contributions in their projects.
+This may explicitly include proprietary projects.
+We want to enable all contributors to be able to use VOLK wherever they want.
+At the same time, we want to make sure that improvements to VOLK itself are easily obtained by everyone, i.e. strike a balance between permissiveness and strict copyleft.
+Previously, several attemps were made to relicense VOLK without success.
+With a formal process and a clear vision laid out, we hope to be able to get everyone on board.
+
+Since VOLK is a library it should choose a fitting license.
+If we see greater adoption of VOLK in more project, we hope to also see more contributions.
+May it be bug fixes, new kernels or even contributions to core features.
+
+
+## Description
+
+In order to facilitate this relicensing effort, we need to got through several steps.
+
+
+### Notification
+
+We consider it to be very important to let every contributor know that the wish to relicense by some contributors exists.
+A first step would be to notify every contributor of this GREP.
+
+
+### The proposed license
+
+We propose [LGPLv3+](https://www.gnu.org/licenses/lgpl-3.0.html) for the new license.
+LGPLv3+ strikes a balance between permissiveness and obligation to provide modified source code.
+All modifications to VOLK should be made available under the same terms under which everyone can use VOLK.
+This should not require users to share code that uses VOLK unaltered.
```

Comment:
It does, practically speaking, make it hard/impossible to use VOLK in statically compiled proprietary projects. But I still think LGPL3 is the right choice!

Reply:
The whole "static vs dynamic linking" license discussion is confusing. Are there any good sources on the subject? Something we can refer to?

Reply:
The general idea is that if you ship something using lgpl code, the user has to have a way to replace that code by a new  / updated / modified version of it.  If it's provided as a dynamic library, it's easy to do ... if it's link statically, you'd need to provide binary object before the final linking along with any linking script / tool to allow the user to redo the final link with a modified version of the library.